criu/cr-restore.c

```diff
@@ -3777,6 +3777,15 @@ static int sigreturn_restore(pid_t pid, struct task_restore_args *task_args, uns
 	 */
 	task_args->lsm_type = kdat.lsm;
 
+	/*
+	 * Decide restore with `sys_preadv` or `sys_mmap`,
+	 * sometimes we don't want to load all memory at once.
+	 */
+	if (opts.restore_on_demand)
+		task_args->on_demand_restore = true;
+	else
+		task_args->on_demand_restore = false;
```

Comment:
`task_args->on_demand_restore = opts.restore_on_demand;`

Reply:
This statement is redundant, and should be simplified.